bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableInstanceGrpcClient.java

```diff
@@ -89,23 +106,20 @@ public void waitForOperation(Operation operation) throws IOException {
                 "System returned invalid response for Operation check: " + currentOperationState);
         }
       }
-      final long waitMs;
-      long timePassedS = TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis() - startMs);
-      if (timePassedS < 5) {
-        waitMs = 250;
-      } else if (timePassedS < 60){
-        waitMs = 1000;
-      } else if (timePassedS < 300) {
-        waitMs = 10000;
-      } else {
-        waitMs = 60000;
-      }
+
+      final long backOffMillis;
       try {
-        Thread.sleep(waitMs);
-      } catch (InterruptedException e) {
-        Thread.currentThread().interrupt();
-        throw new IOException("Waiting for operation was interrupted.", e);
+        backOffMillis = backOff.nextBackOffMillis();
+      } catch (IOException e) {
+        //Will never happen
+        throw new RuntimeException(e);
       }
+      if (backOffMillis == BackOff.STOP) {
+        throw new TimeoutException("Operation did not complete in time");
+      } else {
+        Thread.sleep(backOffMillis);
```

Comment:
Catch InterruptedException? Do you expect caller to handle InterruptedException?